src/iuse.cpp

```diff
@@ -2104,6 +2112,18 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
     // Get the id of the material used
     const auto &repair_item = mod_materials[choice];
 
+    /*
+    DAMAGE+DESTROY
+    WASTED
+    MODIFY_WASTED
+    MODIFY
+    */
+
+    bool consume_mats = true;
+    bool consume_reqs = true;
```

Comment:
`consume_reqs` is set to true only in one case - when you damage item.

        p->add_msg_if_player(m_bad, _("You damage your %s trying to modify it and also waste %d filament charges!"),